student/tests.py

```diff
@@ -105,3 +118,33 @@ def test_get_student_raises_exception_for_non_student_user():
     example_user_not_student.save()
     with pytest.raises(Student.DoesNotExist):
         Student.get_student(example_user_not_student.id)
+
+
+@pytest.mark.django_db
+def test_update_student_account_settings(example_student_a, example_student_a_data_as_dictionary):
```

Comment:
This is not really testing the user flow - only the form object is being tested and that does not really contain much logic in it.

The right way to test is to consider the various flows through the form view:

1.   When the form is initially fetched, populated with initial data from the DB and displayed.
2.  When form data is posted from the browser and its invalid
3.  When form data is posted from the browser and its valid

And write a test for each that used the `client` fixture to emulate the browser activity.

Reply:
@guyy13 you can assert the status code that is returned from the POST request

Reply:
You check the form object in the context. Generally, since there is not real logic in the object - just the definition of its relationship with the model object, its may be enough to test its type to make sure the right one s displayed, but you can also check the the form object is associated with the right model object.

In the case of validation testing, you can inspect the `errors` property on the form object or call the `is_valid()` method.

See the reference for Form objects:
https://docs.djangoproject.com/en/4.0/ref/forms/api/#django.forms.Form

Reply:
@ifireball 
I managed to get the form from the context of the page and test the type of it as you suggested, but I don't understand how to check that the fields in the form are filled out correctly with the request's user details, can you please clarify?
Thanks.
Here's my code currently:
![image](https://user-images.githubusercontent.com/91069555/146370518-a2359241-15d9-42cc-8777-0bf30ea9e02c.png)

Reply:
Ok nvm, I think I managed to do it using `form.initial`

Reply:
I think I managed to implement the tests, please take a look.